pkg/sql/sem/builtins/builtins.go

```diff
@@ -9629,3 +9736,26 @@ func prettyStatement(p tree.PrettyCfg, stmt string) (string, error) {
 	}
 	return formattedStmt.String(), nil
 }
+
+func floatToTimestampTZ(ts float64) (tree.Datum, error) {
+	if math.IsNaN(ts) {
+		return nil, pgerror.New(pgcode.DatetimeFieldOverflow, "timestamp cannot be NaN")
+	}
+	if ts == math.Inf(0) {
+		return tree.MakeDTimestampTZ(pgdate.TimeInfinity, time.Microsecond)
+	}
+	if ts == math.Inf(-1) {
+		return tree.MakeDTimestampTZ(pgdate.TimeNegativeInfinity, time.Microsecond)
+	}
+	strts := strings.Split(strconv.FormatFloat(ts, 'f', 6, 64), ".")
```

Comment:
we should not be casting to string to do this.
return tree.MakeDTimestampTZ(timeutil.Unix(0, int64(ts * float64(time.Second)), time.Microsecond)` should do the trick!

Reply:
Using `tree.MakeDTimestampTZ(timeutil.Unix(0, int64(ts * float64(time.Second)), time.Microsecond)` instead of custing to string.